iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/GpuParquetReaderPostProcessor.scala

```diff
@@ -69,8 +70,10 @@ class GpuParquetReaderPostProcessor(
       s"File read schema field count ${fileReadSchema.getFieldCount} doesn't match expected " +
         s"columnar batch columns ${originalBatch.numCols()}")
 
-    withResource(new ColumnarBatchHandler(this, originalBatch)) { handler =>
-      TypeUtil.visit(expectedSchema, handler).left.get
+    withRetryNoSplit(GpuColumnVector.incRefCounts(originalBatch)) { batch =>
```

Comment:
There is a lot going on here and it looks like it is violating the requirements for retry to work properly. All data that goes into a retry must be spillable. That is so that if we do need to pause a thread or rollback all of that data can be spilled.

originalBatch is not spilliable, what is more you are incrementing the reference counts to it so it cannot be spilled. Why are you doing this?